message.go

```diff
@@ -59,6 +59,29 @@ func MarshalMessage(m Message) ([]byte, error) {
 	return im.Marshal(nil)
 }
 
+// MarshalMessageChecksum marshals a Message into its binary form and prepends
+// an ICMPv6 message with the correct type.
+func MarshalMessageChecksum(m Message, source, destination net.IP) ([]byte, error) {
+	mb, err := m.marshal()
+	if err != nil {
+		return nil, err
+	}
+
+	im := icmp.Message{
+		Type: m.Type(),
+		// Always zero.
+		Code: 0,
+		// Calculated by call to Marshal below
+		Checksum: 0,
+		Body: &icmp.DefaultMessageBody{
+			Data: mb,
+		},
+	}
+
+	// Pseudo-header always nil so checksum is calculated by caller or OS.
```

Comment:
Nit: this comment is no longer true, but can be updated when you refactor this body into an unexported function.

Reply:
doh! I thought I removed that. 😳